docker-images/kafka/scripts/kafka_config_generator.sh

```diff
@@ -2,278 +2,40 @@
 
 # Parameters:
 # $1: Broker ID
-function get_address_for_broker {
-  for ADDRESS in $KAFKA_EXTERNAL_ADDRESSES ; do
-    if [[ $ADDRESS == $1://* ]] ; then
-      echo ${ADDRESS#"$1://"}
+# $2: List of options
+function get_option_for_broker {
+  for OPTION in $2 ; do
+    if [[ $OPTION == $1://* ]] ; then
+      echo ${OPTION#"$1://"}
     fi
   done
 }
 
-#####
-# REPLICATION listener
-#####
-LISTENERS="REPLICATION://0.0.0.0:9091"
-ADVERTISED_LISTENERS="REPLICATION://$(hostname -f):9091"
-LISTENER_SECURITY_PROTOCOL_MAP="REPLICATION:SSL"
-
-#####
-# PLAIN listener
-#####
-if [ "$KAFKA_CLIENT_ENABLED" = "TRUE" ]; then
-  LISTENERS="${LISTENERS},CLIENT://0.0.0.0:9092"
-  ADVERTISED_LISTENERS="${ADVERTISED_LISTENERS},CLIENT://$(hostname -f):9092"
-
-  if [ "$KAFKA_CLIENT_AUTHENTICATION" = "scram-sha-512" ]; then
-    LISTENER_SECURITY_PROTOCOL_MAP="${LISTENER_SECURITY_PROTOCOL_MAP},CLIENT:SASL_PLAINTEXT"
-    CLIENT_LISTENER=$(cat <<EOF
-# CLIENT listener authentication
-listener.name.client.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required;
-listener.name.client.sasl.enabled.mechanisms=SCRAM-SHA-512
-EOF
-)
-  elif [ "$KAFKA_CLIENT_AUTHENTICATION" = "oauth" ]; then
-    LISTENER_SECURITY_PROTOCOL_MAP="${LISTENER_SECURITY_PROTOCOL_MAP},CLIENT:SASL_PLAINTEXT"
-
-    if [ ! -z "$STRIMZI_CLIENT_OAUTH_CLIENT_SECRET" ]; then
-      CLIENT_OAUTH_SECRET="oauth.client.secret=\"$STRIMZI_CLIENT_OAUTH_CLIENT_SECRET\""
-    fi
-
-    if [ -f "/tmp/kafka/oauth-client.truststore.p12" ]; then
-      CLIENT_OAUTH_TRUSTSTORE="oauth.ssl.truststore.location=\"/tmp/kafka/oauth-client.truststore.p12\" oauth.ssl.truststore.password=\"${CERTS_STORE_PASSWORD}\" oauth.ssl.truststore.type=\"PKCS12\""
-    fi
-
-    CLIENT_LISTENER=$(cat <<EOF
-# CLIENT listener authentication
-listener.name.client.oauthbearer.sasl.server.callback.handler.class=io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler
-listener.name.client.oauthbearer.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub="thePrincipalName" ${STRIMZI_CLIENT_OAUTH_OPTIONS} ${CLIENT_OAUTH_SECRET} ${CLIENT_OAUTH_TRUSTSTORE};
-listener.name.client.sasl.enabled.mechanisms=OAUTHBEARER
-EOF
-)
-  else
-    LISTENER_SECURITY_PROTOCOL_MAP="${LISTENER_SECURITY_PROTOCOL_MAP},CLIENT:PLAINTEXT"
-  fi
+# get broker rack if it's enabled
```

Comment:
Can we mention (for the benefit of other contributors) that this file gets created by the init container?